### PR TITLE
[PATCH API-NEXT v2] SHA-384 and AES-XCBC-MAC

### PR DESCRIPTION
**Pull request by lumag**
Extract API definitions from #422 to let them be merged before API freeze. Work on #422 can continue to be merged when it is ready.

### CODE REVIEW

**Comment by lumag** [comment]
@psavol @muvarov as promised, this is API changes split from #422

**Comment by lumag** [comment]
@muvarov somehow this PR received 'No_Email_sent' label and no emails were generated. Could you please fix the label?

File: include/odp/api/spec/crypto.h [hunk 1]
```diff
@@ -143,13 +143,19 @@ typedef enum {
...
+	ODP_AUTH_ALG_SHA384_HMAC,
+
 	/** HMAC-SHA-512
 	 *
 	 *  SHA-512 algorithm in HMAC mode
 	 */
 	ODP_AUTH_ALG_SHA512_HMAC,
 
-	/** AES-GCM
+	/** AES in Galois/Counter Mode
```
Comment (psavol) [inline include/odp/api/spec/crypto.h]: Should not change the heading for the algorithm. I'm using algorithm short names from specifications. E.g. RFC 4106 specifies the name like this.

"This document describes the use of AES in GCM mode (AES-GCM) as an
   IPsec ESP mechanism for confidentiality and data origin
   authentication.  We refer to this method as AES-GCM-ESP." 

And the section describing GCM is:
"2.  AES-GCM"

Comment (lumag) [reply include/odp/api/spec/crypto.h]: Missed this change. Should be fixed now.

**Comment by muvarov** [comment]
Merged.

**Pull request closed without merging**